modules/web-console/src/main/java/org/apache/ignite/console/WebConsoleServer.java

```diff
@@ -142,21 +176,49 @@ public WebConsoleServer(
         router.route().handler(SessionHandler.create(LocalSessionStore.create(vertx)));
         router.route().handler(UserSessionHandler.create(auth));
 
+        if (!F.isEmpty(cfg.getWebRoot()))
+            router.route().handler(StaticHandler.create(cfg.getWebRoot()));
+
         router.route("/eventbus/*").handler(sockJsHnd);
 
         registerRestRoutes(router);
 
         registerVisorTasks();
 
-        // Start HTTP server for browsers and web agents.
         HttpServerOptions httpOpts = new HttpServerOptions()
             .setCompressionSupported(true)
             .setPerMessageWebsocketCompressionSupported(true);
 
+        if (!F.isEmpty(cfg.getKeyStore()) || !F.isEmpty(cfg.getTrustStore())) {
+            httpOpts.setSsl(true);
```

Comment:
HTTPS also required redirect from HTTP to HTTPS.
From old backend  `const sslOptions = { enable301Redirects: true,`

Reply:
Fixed